test/models/user_test.rb

```diff
@@ -0,0 +1,45 @@
+require "test_helper"
+
+describe User do
+  describe "Validations" do
+    let(:user) { User.new }
+
+    it "must be invalid when fields are left blank" do
+      value(user.valid?).must_equal false
+    end
+
+    it "is valid when username is provided" do
+      user1 = users(:user1)
+
+      value(user1.valid?).must_equal true
+    end
+
+    it "is invalid if the username is not unique" do
+      user1 = users(:user1)
+      user2 = User.create(username: "Vinnie")
+
+      value(user1.valid?).must_equal true
+      value(user2.valid?).must_equal false
+    end
+
+    it "is invalid if the username is not between 5 and 20 characters" do
```

Comment:
You should test the username at 4 letters and then 5, verifying that one is invalid and the other valid.  Then verify that 20 characters are valid and 21 invalid.  

Check the edge cases.